test/e2e/common_suite_test.go

```diff
@@ -73,38 +101,69 @@ func (tc *testCase) run() {
 					t.Fatal(err)
 				}
 			}
-
-			if err = client.Resources().Create(ctx, tc.pod); err != nil {
-				t.Fatal(err)
+			if tc.job != nil {
+				if err = client.Resources().Create(ctx, tc.job); err != nil {
+					t.Fatal(err)
+				}
+				if err = wait.For(conditions.New(client.Resources()).JobCompleted(tc.job), wait.WithTimeout(WAIT_JOB_RUNNING_TIMEOUT)); err != nil {
+					t.Log(err)
+				}
 			}
-
-			if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
-				t.Fatal(err)
+			if tc.pod != nil {
+				if err = client.Resources().Create(ctx, tc.pod); err != nil {
+					t.Fatal(err)
+				}
+				if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
+					t.Fatal(err)
+				}
 			}
-
 			return ctx
 		}).
 		Assess(tc.assessMessage, func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
-			tc.assert.HasPodVM(t, tc.pod.Name)
-
+			client := cfg.Client()
 			var podlist v1.PodList
 
-			if err := cfg.Client().Resources(tc.pod.Namespace).List(context.TODO(), &podlist); err != nil {
-				t.Fatal(err)
-			}
-
-			for _, testCommand := range tc.testCommands {
-				var stdout, stderr bytes.Buffer
-
-				for _, podItem := range podlist.Items {
-					if podItem.ObjectMeta.Name == tc.pod.Name {
-						if err := cfg.Client().Resources(tc.pod.Namespace).ExecInPod(ctx, tc.pod.Namespace, tc.pod.Name, testCommand.containerName, testCommand.command, &stdout, &stderr); err != nil {
-							t.Log(stderr.String())
-							t.Fatal(err)
-						}
+			if tc.job != nil {
+				if err := client.Resources(tc.job.Namespace).List(ctx, &podlist); err != nil {
+					t.Fatal(err)
+				}
+				successPod, errorPod, podLogString, err := getSuccessfulAndErroredPods(ctx, t, client, *tc.job)
+				if err != nil {
+					t.Fatal(err)
+				}
+				if errorPod == len(podlist.Items) && successPod == 0 {
+					t.Errorf("Job Failed to Start pod")
+				}
+				if successPod == 1 && errorPod >= 1 {
```

Comment:
I think while running this current job , it will create only one pod with `Completed` status also it Fails multiple pods sometimes to reach `Completed` status. Thus in the end we are expecting only one Pod to reach `Completed` status.
Please Correct me if I'm wrong.